### PR TITLE
Update index BinarySet key name to be compatible with knowhere-1.x

### PR DESCRIPTION
**Pull request by cydrain**

### CODE REVIEW

File: src/index/flat/flat.cc [hunk 1]
```diff
@@ -265,10 +265,12 @@ class FlatIndexNode : public IndexNode {
             }
             std::shared_ptr<uint8_t[]> data(writer.data_);
             if constexpr (std::is_same<T, faiss::IndexFlat>::value) {
-                binset.Append("FLAT", data, writer.rp);
+                // use "IVF" to be compatible with knowhere-1.x
+                binset.Append("IVF", data, writer.rp);
```
Comment (liliu-z) [inline src/index/flat/flat.cc:269]: For backward compatibility we can support both `FLAT` and `IVF` as Flat Index for `Deserialization` and only support `FLAT` for Serialization

**Review by liliu-z — CHANGES_REQUESTED** [review]

File: src/index/flat/flat.cc (hunk 1 above)
Comment (cydrain) [reply src/index/flat/flat.cc]: updated

**Pull request merged**